l1-contracts/src/governance/GSE.sol

```diff
@@ -87,9 +90,32 @@ interface IGSE is IGSECore {
     view
     returns (address);
   function getPowerUsed(address _delegatee, uint256 _proposalId) external view returns (uint256);
-  function getCanonicalMagicAddress() external view returns (address);
+  function getBonusInstanceAddress() external view returns (address);
 }
 
+/**
+ * @title GSECore
+ * @author Aztec Labs
+ * @notice The core Governance Staking Escrow contract that handles the staking of attesters on rollup instances.
+ *         It is responsible for:
+ *         - depositing/withdrawing attesters on rollup instances
+ *         - providing rollup instances with historical views of their attesters
+ *         - allowing depositors to delegate their voting power
+ *         - allowing delegatees to vote at governance
+ *         - maintaining a set of "bonus" attesters which are always staked on the latest rollup
```

Comment:
Would extend a little on this to convey the implications, e.g., automatically moving along etc. Since it might otherwise be missed by the reader.